src/node_nanomsg.cc

```diff
@@ -435,130 +434,17 @@ void InitAll(Handle<Object> exports) {
-    // positive for socket types.
-    EXPORT_CONSTANT(exports, NN_SOL_SOCKET);
-
-    //  Generic socket options (NN_SOL_SOCKET level).
-    EXPORT_CONSTANT(exports, NN_LINGER);
-    EXPORT_CONSTANT(exports, NN_SNDBUF);
-    EXPORT_CONSTANT(exports, NN_RCVBUF);
-    EXPORT_CONSTANT(exports, NN_SNDTIMEO);
-    EXPORT_CONSTANT(exports, NN_RCVTIMEO);
-    EXPORT_CONSTANT(exports, NN_RECONNECT_IVL);
-    EXPORT_CONSTANT(exports, NN_RECONNECT_IVL_MAX);
-    EXPORT_CONSTANT(exports, NN_SNDPRIO);
-    EXPORT_CONSTANT(exports, NN_SNDFD);
-    EXPORT_CONSTANT(exports, NN_RCVFD);
-    EXPORT_CONSTANT(exports, NN_DOMAIN);
-    EXPORT_CONSTANT(exports, NN_PROTOCOL);
-    EXPORT_CONSTANT(exports, NN_IPV4ONLY);
-    EXPORT_CONSTANT(exports, NN_SOCKET_NAME);
-
-    // transport options
-    EXPORT_CONSTANT(exports, NN_SUB_SUBSCRIBE);
-    EXPORT_CONSTANT(exports, NN_SUB_UNSUBSCRIBE);
-    EXPORT_CONSTANT(exports, NN_REQ_RESEND_IVL);
-    EXPORT_CONSTANT(exports, NN_SURVEYOR_DEADLINE);
-    EXPORT_CONSTANT(exports, NN_TCP_NODELAY);
-
-    // Flags
-    EXPORT_CONSTANT(exports, NN_DONTWAIT);
-
-    // Errors
-    EXPORT_CONSTANT(exports, EADDRINUSE);
-    EXPORT_CONSTANT(exports, EADDRNOTAVAIL);
-    EXPORT_CONSTANT(exports, EAFNOSUPPORT);
-    EXPORT_CONSTANT(exports, EAGAIN);
-    EXPORT_CONSTANT(exports, EBADF);
-    EXPORT_CONSTANT(exports, ECONNREFUSED);
-    EXPORT_CONSTANT(exports, EFAULT);
-    EXPORT_CONSTANT(exports, EFSM);
-    EXPORT_CONSTANT(exports, EINPROGRESS);
-    EXPORT_CONSTANT(exports, EINTR);
-    EXPORT_CONSTANT(exports, EINVAL);
-    EXPORT_CONSTANT(exports, EMFILE);
-    EXPORT_CONSTANT(exports, ENAMETOOLONG);
-    EXPORT_CONSTANT(exports, ENETDOWN);
-    EXPORT_CONSTANT(exports, ENOBUFS);
-    EXPORT_CONSTANT(exports, ENODEV);
-    EXPORT_CONSTANT(exports, ENOMEM);
-    EXPORT_CONSTANT(exports, ENOPROTOOPT);
-    EXPORT_CONSTANT(exports, ENOTSOCK);
-    EXPORT_CONSTANT(exports, ENOTSUP);
-    EXPORT_CONSTANT(exports, EPROTO);
-    EXPORT_CONSTANT(exports, EPROTONOSUPPORT);
-    EXPORT_CONSTANT(exports, ETERM);
-    EXPORT_CONSTANT(exports, ETIMEDOUT);
-    EXPORT_CONSTANT(exports, EACCES);
-    EXPORT_CONSTANT(exports, ECONNABORTED);
-    EXPORT_CONSTANT(exports, ECONNRESET);
-    EXPORT_CONSTANT(exports, EHOSTUNREACH);
-    EXPORT_CONSTANT(exports, EMSGSIZE);
-    EXPORT_CONSTANT(exports, ENETRESET);
-    EXPORT_CONSTANT(exports, ENETUNREACH);
-    EXPORT_CONSTANT(exports, ENOTCONN);
-    EXPORT_CONSTANT(exports, EISCONN); // not in symbol.c, but is in nn.h
-
-    // Polling - these aren't in symbol.c but they are in nn.h
-    EXPORT_CONSTANT(exports, NN_POLLIN);
-    EXPORT_CONSTANT(exports, NN_POLLOUT);
+    // Export symbols.
+    int index = 0;
+    int value;
+    while (true) {
+        const char *symbol_name = nn_symbol(index, &value);
+        if (symbol_name == NULL) {
+            break;
+        }
+        exports->Set(NanNew(symbol_name), NanNew<Number>(value));
+        index += 1;
+    }
```

Comment:
Can we turn this into a for-loop, use pre-increment on the loop variable, and put `int value` in the for-loop's block?